x-pack/plugins/apm/public/components/app/service_overview/index.tsx

```diff
@@ -77,124 +52,21 @@ export function ServiceOverview() {
-                </EuiFlexItem>
-              )}
-              <EuiFlexItem grow={7}>
-                <EuiPanel hasBorder={true}>
-                  <ServiceOverviewErrorsTable serviceName={serviceName} />
-                </EuiPanel>
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          </EuiFlexItem>
-          <EuiFlexItem>
-            <EuiFlexGroup
-              direction={rowDirection}
-              gutterSize="s"
-              responsive={false}
-            >
-              {isServerless ? (
-                <EuiFlexItem grow={3}>
-                  <TransactionColdstartRateChart
-                    showAnnotations={false}
-                    environment={environment}
-                    kuery={kuery}
-                  />
-                </EuiFlexItem>
-              ) : (
-                <EuiFlexItem grow={3}>
-                  <TransactionBreakdownChart
-                    showAnnotations={false}
-                    environment={environment}
-                    kuery={kuery}
-                  />
-                </EuiFlexItem>
-              )}
-              {!isRumAgent && (
-                <EuiFlexItem grow={7}>
-                  <EuiPanel hasBorder={true}>
-                    <ServiceOverviewDependenciesTable
-                      fixedHeight={true}
-                      showPerPageOptions={false}
-                      link={
-                        <EuiLink href={dependenciesLink}>
-                          {i18n.translate(
-                            'xpack.apm.serviceOverview.dependenciesTableTabLink',
-                            { defaultMessage: 'View dependencies' }
-                          )}
-                        </EuiLink>
-                      }
-                    />
-                  </EuiPanel>
-                </EuiFlexItem>
-              )}
-            </EuiFlexGroup>
-          </EuiFlexItem>
-          {!isRumAgent && !isMobileAgent && !isServerless && (
-            <EuiFlexItem>
-              <EuiFlexGroup
-                direction="column"
-                gutterSize="s"
-                responsive={false}
-              >
-                <ServiceOverviewInstancesChartAndTable
-                  chartHeight={nonLatencyChartHeight}
-                  serviceName={serviceName}
-                />
-              </EuiFlexGroup>
-            </EuiFlexItem>
-          )}
-        </EuiFlexGroup>
+        {!isMobileAgent ? (
+          <ServiceOverviewCharts
+            latencyChartHeight={latencyChartHeight}
+            rowDirection={rowDirection}
+            nonLatencyChartHeight={nonLatencyChartHeight}
+            isSingleColumn={isSingleColumn}
+          />
+        ) : (
+          <ServiceOverviewMobileAgentCharts
+            latencyChartHeight={latencyChartHeight}
+            rowDirection={rowDirection}
+            nonLatencyChartHeight={nonLatencyChartHeight}
+            isSingleColumn={isSingleColumn}
+          />
+        )}
```

Comment:
I think it's better to have the truth first in this case:

```suggestion
        {isMobileAgent ? (
          <ServiceOverviewMobileAgentCharts
            latencyChartHeight={latencyChartHeight}
            rowDirection={rowDirection}
            nonLatencyChartHeight={nonLatencyChartHeight}
            isSingleColumn={isSingleColumn}          
          />
        ) : (
        <ServiceOverviewCharts
            latencyChartHeight={latencyChartHeight}
            rowDirection={rowDirection}
            nonLatencyChartHeight={nonLatencyChartHeight}
            isSingleColumn={isSingleColumn}
          />
        )}
```

Reply:
And since they share the same props you could create a variable and then do something like:

```suggestion
        {!isMobileAgent ? (
          <ServiceOverviewCharts {...props} />
        ) : (
          <ServiceOverviewMobileAgentCharts  {...props}  />
        )}
```

Reply:
Makes sense, updated in https://github.com/elastic/kibana/pull/143697/commits/9cab05301c4352db61467fd0923f573308443f48